frontend/src/metabase/common/components/DataPicker/hooks/useAvailableData.ts

```diff
@@ -6,11 +6,16 @@ interface Props {
 }
 
 export const useAvailableData = ({ databaseId }: Props = {}) => {
-  const { data } = useSearchQuery({
-    limit: 0,
-    models: ["card"],
-    table_db_id: databaseId,
-  });
+  const { data } = useSearchQuery(
+    {
+      limit: 0,
+      models: ["card"],
+      table_db_id: databaseId,
+    },
+    {
+      refetchOnMountOrArgChange: true,
```

Comment:
This is the fix

Reply:
Can this same approach be used to fix this?
https://github.com/metabase/metabase/issues/44223

Reply:
> Can this same approach be used to fix this? #44223

I didn't investigate that issue, but probably not, because `useListRecentsQuery` in `EntityPickerModal` already uses `refetchOnMountOrArgChange: true`